tests/rational.rs

```diff
@@ -126,12 +136,29 @@ impl SynthLanguage for Math {
         let solver = z3::Solver::new(&ctx);
         let lexpr = egg_to_z3(&ctx, Self::instantiate(lhs).as_ref());
         let rexpr = egg_to_z3(&ctx, Self::instantiate(rhs).as_ref());
-        solver.assert(&lexpr._eq(&rexpr).not());
-        match solver.check() {
-            z3::SatResult::Unsat => ValidationResult::Valid,
-            z3::SatResult::Unknown => ValidationResult::Unknown,
-            z3::SatResult::Sat => ValidationResult::Invalid,
+        let lhs_denom = Self::all_denominators(Self::pat_to_sexp(lhs));
+        let rhs_denom = Self::all_denominators(Self::pat_to_sexp(rhs));
+        let denominators = lhs_denom.union(&rhs_denom);
+
+        let mut assert_equal = lexpr._eq(&rexpr);
+        let zero_pat = "0".parse::<Pattern<Math>>().unwrap();
+        let zero_z3 = egg_to_z3(&ctx, Self::instantiate(&zero_pat).as_ref());
```

Comment:
I think `let zero = z3::ast::Real::from_real(ctx, 0, 1);` would be an easier way to define zero